extensions/indexes/range/src/main/java/org/exist/indexing/range/RangeIndexConfigAttributeCondition.java

```diff
@@ -56,143 +56,136 @@ public class RangeIndexConfigAttributeCondition extends RangeIndexConfigConditio
 
         if (parentPath.getLastComponent().getNameType() == ElementValue.ATTRIBUTE) {
-            throw new DatabaseConfigurationException("Range index module: Attribute condition cannot be defined for an attribute:" + parentPath.toString());
+            throw new DatabaseConfigurationException(
+                    "Range index module: Attribute condition cannot be defined for an attribute:" + parentPath);
         }
 
-        this.attributeName = elem.getAttribute("attribute");
-        if (this.attributeName == null || this.attributeName.isEmpty()) {
+        attributeName = elem.getAttribute("attribute");
+        if (attributeName == null || attributeName.isEmpty()) {
             throw new DatabaseConfigurationException("Range index module: Empty or no attribute qname in condition");
         }
 
         try {
-            this.attribute = new QName(QName.extractLocalName(this.attributeName), XMLConstants.NULL_NS_URI, QName.extractPrefix(this.attributeName), ElementValue.ATTRIBUTE);
+            attribute = new QName(QName.extractLocalName(attributeName), XMLConstants.NULL_NS_URI,
+                    QName.extractPrefix(attributeName), ElementValue.ATTRIBUTE);
         } catch (final QName.IllegalQNameException e) {
             throw new DatabaseConfigurationException("Rand index module error: " + e.getMessage(), e);
         }
-        this.value = elem.getAttribute("value");
-
+        value = elem.getAttribute("value");
 
         // parse operator (default to 'eq' if missing)
         if (elem.hasAttribute("operator")) {
             final String operatorName = elem.getAttribute("operator");
-            this.operator = Operator.getByName(operatorName.toLowerCase());
-            if (this.operator == null) {
-                throw new DatabaseConfigurationException("Range index module: Invalid operator specified in range index condition: " + operatorName + ".");
+            operator = Operator.getByName(operatorName.toLowerCase());
+            if (operator == null) {
+                throw new DatabaseConfigurationException(
+                        "Range index module: Invalid operator specified in range index condition: " + operatorName + ".");
             }
         } else {
-            this.operator = Operator.EQ;
+            operator = Operator.EQ;
         }
 
-
         final String caseString = elem.getAttribute("case");
-        final String numericString = elem.getAttribute("numeric");
+        caseSensitive = (caseString != null && !caseString.equalsIgnoreCase("no"));
 
-        this.caseSensitive = (caseString != null && !caseString.equalsIgnoreCase("no"));
-        this.numericComparison = (numericString != null && numericString.equalsIgnoreCase("yes"));
+        final String numericString = elem.getAttribute("numeric");
+        numericComparison = (numericString != null && numericString.equalsIgnoreCase("yes"));
```

Comment:
I tend to use StringUtils here for readability... but that is another dependency.

https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#equalsIgnoreCase-java.lang.CharSequence-java.lang.CharSequence-

Reply:
Since this pattern is littered throughout exist-db's code-base it would either make sense to:
- create our own utility function
- use string-utils everywhere